nasa/run.sh

```diff
@@ -1,30 +1,41 @@
-#!/bin/bash
-# This is intended for running DPS jobs.
-
-# The environment.yaml file sets up a custom conda environment called `dem`.
-# For NASA MAAP DPS, use `source activate <custom>`, not `conda activate <custom>`
-source activate dem
+#!/usr/bin/env bash
 
-# [left  bottom  right top]
-INPUT_LEFT=$1
-INPUT_BOTTOM=$2
-INPUT_RIGHT=$3
-INPUT_TOP=$4
-COMPUTE=$5
+set -xeou pipefail
 
-# Get path to this run.sh script
-basedir=$( cd "$(dirname "$0")" ; pwd -P )
-REPO_ROOT_PATH=$(dirname ${basedir})
+# This is intended for running DPS jobs.
 
-# Per NASA MAAP DPS convention, all outputs MUST be placed
-# by the algorithm into a directory called "output".
-# Once the DPS job finishes, MAAP will copy everything from "output"
-# to a directory in my-public-bucket. Everything else on the instance
-# will be destroyed.
-mkdir -p ${PWD}/output
+# `bbox` is expected to be 4 space-separated coords: left bottom right top
+bbox=$1
+# `compute` is optional, but any non-empty value will be converted to `--compute`
+compute=${2:+--compute}
 
-# Setup the environment variables. (Req'd for sardem)
-export HOME=/home/ops
+# path to the base of this repo (one level up from this script)
+basedir=$(dirname "$(dirname "$(readlink -f "$0")")")
 
-python ${REPO_ROOT_PATH}/get_dem.py --bbox ${INPUT_LEFT} ${INPUT_BOTTOM} ${INPUT_RIGHT} ${INPUT_TOP} ${COMPUTE} --out_dir ${PWD}/output
+# Per NASA MAAP DPS convention, all outputs MUST be written under a directory
+# named 'output' relative to the current working directory.  Once the DPS job
+# finishes, MAAP will copy everything from 'output' to a directory under
+# 'my-private-bucket/dps_output'. Everything else on the instance will be lost.
+outdir="${PWD}/output"
```

Comment:
The code in `get_dem.py` makes the directory. That way, if `get_dem.py` is executed directly (instead of via `run.sh`, it creates whatever directory is specified for the `--out_dir` option).

Reply:
@arthurduf , Does the ESA MAAP have any specific needs for where/how the output directory is created when running on DPS? Thanks!

Reply:
The output directory ( and the working direction) needs to be in a specific folder ( /projects/data). 
 
There is no need to create it before launch if the the process creates it....but by habit I create every needed folder in the entrypoint of the container